api/src/main/java/org/openmrs/api/impl/ConditionServiceImpl.java

```diff
@@ -101,46 +103,53 @@ public List<Condition> getConditionsByEncounter(Encounter encounter) throws APIE
 	 */
 	@Override
 	public Condition saveCondition(Condition condition) throws APIException {
-		Condition existingCondition = Context.getConditionService().getConditionByUuid(condition.getUuid());
+		
 		// If there is no existing condition, then we are creating a condition
-		if (existingCondition == null) {
+		Integer existingConditionId = condition.getConditionId();
+		if (existingConditionId == null) {
 			return conditionDAO.saveCondition(condition);
 		}
-
-		// If the incoming condition has been voided, we simply void the existing condition
-		// All other changes are ignored
-		if (condition.getVoided()) {
-			if (!existingCondition.getVoided()) {
-				return Context.getConditionService().voidCondition(existingCondition,
-					StringUtils.isNotBlank(condition.getVoidReason()) ? condition.getVoidReason() : "Condition deleted");
-			} else {
-				return existingCondition;
-			}
+		
+		// If there is an existing condition, create a new condition from it and reset the existing instance state
+		Condition newCondition = Condition.newInstance(condition);
+		Context.refreshEntity(condition);
```

Comment:
what does this do again? loads it in from the DB/cache again? (ie prior to any changes?)